test-menus.py

```diff
@@ -0,0 +1,94 @@
+        """
+        mu = mock_user.return_value
+        mu.can_access_route.return_value=False
+        entries = get_menu_entries(
+            mu, 'Configuration', 'cog', '', [
+                {'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services', 'entries': [{'title': 'Profile mgmt', 'icon': 'bookmark', 'route':
+                    'honeypot.profiles'}]}]}]},
+                {'title': 'Data processing', 'icon': 'exchange', 'route':
+                    'config.data_processing'},
+                {'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services'}
+            ]
+        )
+        correct_entries={}
+
+        self.assertDictEqual(entries,correct_entries)
+
+
+    @patch('mod_auth.models.User')
+    def test_menu_with_partial_permissions(self, mock_user):
+        """
+        Passing a menu entry to the function get_menu_entries() when the user is not
+        allowed to access the route of one of the sub-entries with 'route' = 'honeypot.profiles'
+        This should return a dictionary without this entry.
+        """
+        mu = mock_user.return_value
+
+        def side_effect(*args):
+            if args[0] == 'honeypot.profiles':
+                return False
+            return True
+
+        mu.can_access_route.side_effect = side_effect
+        entries = get_menu_entries(
+            mu, 'Configuration', 'cog', '', [
+                {'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services', 'entries': [{'title': 'Profile mgmt', 'icon': 'bookmark', 'route':
+                    'honeypot.profiles'}]}]}]},
+                {'title': 'Data processing', 'icon': 'exchange', 'route':
+                    'config.data_processing'},
+                {'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services'}
+            ]
+        )
+        correct_entries = {'entries': [{'title': 'Notif. services', 'route': 'config.notifications', 'entries': [{'title': 'Notif. services', 'route': 'config.notifications', 'entries': [{'title': 'Honeypot services', 'route': 'config.services', 'icon': 'sliders'}], 'icon': 'bell-o'}], 'icon': 'bell-o'}, {'route': 'config.data_processing', 'title': 'Data processing', 'icon': 'exchange'}, {'route': 'config.services', 'title': 'Honeypot services', 'icon': 'sliders'}], 'icon': 'cog', 'title': 'Configuration'}
+        self.assertDictEqual(entries, correct_entries)
+
+
+if __name__=='__main__':
```

Comment:
This is fine as it's the only test right now, but this shouldn't be necessary when there's a more global setup of tests

Reply:
Are you talking about line 93 or line 90 ?

Reply:
93. The entire section to run it standalone basically :)

Reply:
Done in the latest commits.